OpenSim/Simulation/Model/Point.h

```diff
@@ -0,0 +1,137 @@
+//=============================================================================
+/**
+ * A Point is an OpenSim abstraction for any point location in space. Points
+ * are intended to locate physical structures (such as points of constraints
+ * and points of muscle attachments) as well as embody the results of spatial
+ * calculations. For example, if your system involves contact, you can define
+ * a Point that describes the location of the center-of-pressure as one
+ * element rolls over another.
+ *
+ * A Point provides its location, velocity and acceleration in the Ground frame
+ * as a function of the Model's (SimTK::MultibodySystem's) state, which is
+ * accessible when the state has been realized to the corresponding
+ * SimTK::Stage (i.e. Position, Velocity and Acceleration). 
+ *
+ * @author Ajay Seth
+ */
+class OSIMSIMULATION_API Point : public ModelComponent {
+OpenSim_DECLARE_ABSTRACT_OBJECT(Point, ModelComponent);
+
+public:
+    OpenSim_DECLARE_OUTPUT(location, SimTK::Vec3, getLocationInGround,
+        SimTK::Stage::Position);
+    OpenSim_DECLARE_OUTPUT(velocity, SimTK::Vec3, getVelocityInGround,
+        SimTK::Stage::Velocity);
+    OpenSim_DECLARE_OUTPUT(acceleration, SimTK::Vec3, getAccelerationInGround,
+        SimTK::Stage::Acceleration);
+    //--------------------------------------------------------------------------
+    // CONSTRUCTION
+    //--------------------------------------------------------------------------
+    Point();
+
+    virtual ~Point() {};
+
+    /** @name Spatial Operations for Point
+    Convenient access to spatial kinematics of Points.*/
+    /**@{**/
+
+    /**
+    Get the location of this Point relative to the Ground frame.
+    @param state       The state applied to the model when determining the
+                       location of the Point.
+    @return location   The location of the point expressed in the Ground. */
+    const SimTK::Vec3& getLocationInGround(const SimTK::State& state) const;
+
+    /** The velocity v_G of this Point expressed in ground.
+        Is only valid at Stage::Velocity or higher.
+    @param state       The state applied to the model when determining the
+                       velocity of the Point.
+    @return velocity   The velocity of the point expressed in the Ground. */
+    const SimTK::Vec3& getVelocityInGround(const SimTK::State& state) const;
+
+    /** The acceleration a_G, of this Point expressed in ground.
+        Is only valid at Stage::Acceleration or higher. 
+    @param state       The state applied to the model when determining the
+                       acceleration of the Point.
+    @return velocity   The acceleration of the point expressed in Ground */
+    const SimTK::Vec3& getAccelerationInGround(const SimTK::State& state) const;
+
+    // End of Point's Spatial Kinematics
+    ///@}
+
+protected:
+    /** @name Point Extension methods.
+    Concrete Point types must override these methods. */
```

Comment:
Can you add a comment that Point takes care of caching the return value of these functions? This is useful to those implementing their own Points, so they know not to do their own caching.